### PR TITLE
Add four tests to cover the following four APIs.

### PR DESCRIPTION
**Pull request by huanwu**
System.Xml.Serialization.XmlDeserializationEvents
System.Xml.Serialization.XmlAttributeEventArgs
System.Xml.Serialization.XmlElementEventArgs
System.Xml.Serialization.XmlNodeEventArgs

### CODE REVIEW

**Comment by huanwu** [comment]
@shmao @mconnew @zhenlan 
#10578 
Not cover System.Xml.Serialization.UnreferencedObjectEventArgs since it is using SOAP, which is blocked by CodeDOM.

**Comment by shmao** [comment]
Tests failed for the following error,

> error CS0246: The type or namespace name 'SoapIncludeAttribute' could not be found (are you missing a using directive or an assembly reference?) [D:\j\workspace\windows_nt_re---ad85658d\src\System.Runtime.Serialization.Json\tests\Performance\System.Runtime.Serialization.Json.Performance.Tests.csproj

**Comment by huanwu** [comment]
@dotnet-bot test Innerloop OSX Release Build and Test, Innerloop OSX Release Build and Test

File: src/System.Private.Xml/tests/XmlSerializer/XmlSerializationTypes.cs [hunk 1]
```diff
@@ -0,0 +1,14 @@
+﻿using System.Xml.Serialization;
```
Comment (shmao) [inline src/System.Private.Xml/tests/XmlSerializer/XmlSerializationTypes.cs]: Please add copyright/License information at the top.

File: src/System.Private.Xml/tests/XmlSerializer/XmlSerializerTests.cs [hunk 2]
```diff
@@ -1887,6 +1888,155 @@ public static void XmlSerializerFactoryTest()
...
+                    <Name>Thing1</Name>
+                    <Elmo>
+                        <Glue>element</Glue>
+                    </Elmo>
+              </coho:ThingInfo>
+  </Group>";
+        count = 0;
+        grouplists = new List<string>();
+        resetevent.Reset();
+        Group group = (Group)serializer.Deserialize(GetStreamFromString(xmlFileContent));
```
Comment (shmao) [inline src/System.Private.Xml/tests/XmlSerializer/XmlSerializerTests.cs]: Can we use `SerializeAndDeserialize` method for these tests?

File: src/System.Private.Xml/tests/XmlSerializer/XmlSerializationTypes.cs [hunk 3]
```diff
@@ -0,0 +1,14 @@
...
+
+[SoapInclude(typeof(Vehicle))]
```
Comment (shmao) [inline src/System.Private.Xml/tests/XmlSerializer/XmlSerializationTypes.cs]: Do the type need this attribute?

File: src/System.Private.Xml/tests/XmlSerializer/XmlSerializerTests.cs [hunk 4]
```diff
@@ -1887,6 +1888,155 @@ public static void XmlSerializerFactoryTest()
...
+    public static void XmlUnknownElementAndEventHandlerTest()
+    {
+        XmlSerializer serializer = new XmlSerializer(typeof(Group));
+        serializer.UnknownElement += new XmlElementEventHandler((o, args) =>
+        {
+            Group myGroup = (Group)args.ObjectBeingDeserialized;
+            Assert.NotNull(myGroup);
+            grouplists.Add(args.Element.Name);
+            ++count;
+            if (count >= 3) resetevent.Set();
```
Comment (shmao) [inline src/System.Private.Xml/tests/XmlSerializer/XmlSerializerTests.cs]: I don't think we need to use AutoResetEvent. We just need to set a flag in the event handler to verify the handler is called.

File: src/System.Private.Xml/tests/XmlSerializer/XmlSerializerTests.cs [hunk 5]
```diff
@@ -1887,6 +1888,155 @@ public static void XmlSerializerFactoryTest()
         Assert.Equal(dog1.Breed, dog2.Breed);
     }
 
+    static List<string> grouplists;
+    static AutoResetEvent resetevent = new AutoResetEvent(false);
+    static int count = 0;
```
Comment (shmao) [inline src/System.Private.Xml/tests/XmlSerializer/XmlSerializerTests.cs]: We don't need these static fields for the tests.

**Comment by stephentoub** [comment]
@dotnet-bot test this please

**Comment by joperezr** [comment]
@huanwu is this ready to merge?

File: src/System.Private.Xml/tests/XmlSerializer/XmlSerializerTests.cs [hunk 6]
```diff
@@ -1887,6 +1888,136 @@ public static void XmlSerializerFactoryTest()
...
+            Assert.NotNull(myGroup);
+            grouplists.Add(args.Element.Name);
+            ++count;
+        });
+        string xmlFileContent = @"<?xml version=""1.0"" encoding=""utf-8""?>
+  <Group xmlns:xsi=""http://www.w3.org/2001/XMLSchema-instance"" xmlns:xsd = ""http://www.w3.org/2001/XMLSchema"">
+         <GroupName>MyGroup</GroupName>
+         <GroupSize>Large</GroupSize>
+         <GroupNumber>444</GroupNumber>
+         <GroupBase>West</GroupBase>
```
Comment (shmao) [inline src/System.Private.Xml/tests/XmlSerializer/XmlSerializerTests.cs]: Can we change these unknown elements' names to something like "UnknownElement1"? I think that would make the test case more clear and self-explanatory.

File: src/System.Private.Xml/tests/XmlSerializer/XmlSerializerTests.cs [hunk 7]
```diff
@@ -1887,6 +1888,136 @@ public static void XmlSerializerFactoryTest()
...
+              <cp:GroupSize>Large</cp:GroupSize>
+              <cp:GroupNumber>444</cp:GroupNumber>
+              <coho:GroupBase>West</coho:GroupBase>
+              <coho:ThingInfo>
+                    <Number>1</Number>
+                    <Name>Thing1</Name>
+                    <Elmo>
+                        <Glue>element</Glue>
+                    </Elmo>
+              </coho:ThingInfo>
```
Comment (shmao) [inline src/System.Private.Xml/tests/XmlSerializer/XmlSerializerTests.cs]: Can we change the unknown nodes' names to "UnknownNode1", "UnknownNode2", and etc.?

**Comment by huanwu** [comment]
@joperezr merged.

**Comment by joperezr** [comment]
@huanwu Thanks :smile:

**Pull request merged**